main.go

```diff
@@ -0,0 +1,90 @@
+///////////////////////////////////////////////////////////////////////
+// Copyright (c) 2018 VMware, Inc. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+///////////////////////////////////////////////////////////////////////
+package main
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"log"
+	"net/http"
+	"os"
+	"os/signal"
+	"strconv"
+
+	"github.com/dispatchframework/funky/pkg/funky"
+)
+
+const (
+	serversEnvVar   = "SERVERS"
+	serverCmdEnvVar = "SERVER_CMD"
+)
+
+type funkyHandler struct {
+	router funky.Router
+}
+
+func (f funkyHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
+	var body funky.Message
+	err := json.NewDecoder(r.Body).Decode(&body)
+	if err != nil {
+		resp := funky.Message{
+			Context: &funky.Context{
+				Error: &funky.Error{
+					ErrorType: funky.InputError,
+					Message:   "Invalid Input",
+				},
+			},
+		}
+		out, _ := json.Marshal(resp)
+		fmt.Fprintf(w, string(out))
+		return
+	}
+
+	resp, _ := f.router.Delegate(body)
+
+	out, _ := json.Marshal(resp)
+
+	fmt.Fprintf(w, string(out))
```

Comment:
Last two lines can be replaced with `json.NewEncoder(w).Encode(resp)` making `out` unnecessary.